numpy/core/src/multiarray/mapping.c

```diff
@@ -2915,20 +2915,20 @@ PyArray_MapIterNew(npy_index_info *indices , int index_num, int index_type,
         Py_INCREF(extra_op_dtype);
         mit->extra_op_dtype = extra_op_dtype;
 
-        /* Create an iterator, just to broadcast the arrays?! */
-        tmp_iter = NpyIter_MultiNew(mit->numiter, index_arrays,
-                                    NPY_ITER_ZEROSIZE_OK |
-                                    NPY_ITER_REFS_OK |
-                                    NPY_ITER_MULTI_INDEX |
-                                    NPY_ITER_DONT_NEGATE_STRIDES,
-                                    NPY_KEEPORDER,
-                                    NPY_UNSAFE_CASTING,
-                                    tmp_op_flags, NULL);
-        if (tmp_iter == NULL) {
-            goto fail;
-        }
-
         if (PyArray_SIZE(subspace) == 1) {
```

Comment:
Weird code ;). One of the uglier parts of my indexing stuff :(. Especially looking at the thing below, I guess this test should actually be an `PyArray_ISONESEGMENT` check. That said, it is also the part I wonder if it really optimizes much (or correctly for all but simple expressions). The idea is to set the output array order to something that is quick to fill I think.

Anyway, looks like a good and safe change in any case, since it just moves obvous branches.

Reply:
well, not sure, and not much time to think about it.Will probably just put it in as is soon (or someone else can do it).